<launch>
  <arg name="bag_file" default="/home/marta/bags/stalker_2022-10-18-12-21-58.bag"/>
  <arg name="rosbag_args" default="-r 6"/>
  <arg name="enable_rviz" default="true"/>
  <arg name="enable_plot" default="true"/>
  <!-- ROSBag -->
  <param name="use_sim_time" value="true"/>
  <node pkg="rosbag" type="play" name="rosbag" args="--clock $(arg rosbag_args) $(arg bag_file)"/>

  <!-- Load robot description -->
  <group ns="sparus2">
    <include file="$(find sparus2_description)/launch/sparus2_description.launch">
      <arg name="robot_name" value="sparus2"/>
    </include>
    
  </group>

  <!-- Upside down TF -->
  <node name="acoustic_loc" pkg="graph_slam_3d" type="acoustic_loc" output="screen">
        <rosparam command="load" file="$(find graph_slam_3d)/config/acoustic_loc.yaml" />
    </node>
    <node if="$(arg enable_plot)" name="graphplot" pkg="graph_slam_3d" type="graphplot.py"/>
  <!-- RViz -->
  <node if="$(arg enable_rviz)" pkg="rviz" type="rviz" name="rviz" args="-d $(find cola2_sparus2)/scene/sparus2.rviz"/>
</launch>
